feat: add a new Widget Segment Radial Gauge, adjust Scale Radial Gauge for negative minValue usage

project2021-code opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

- Added a Segment Radial Gauge which can show different segment color & size
- Edited getScale function of Scale Radial Gauge for negative minValue usage
- Added Example Widget in example/lib/main.dart

**- How I did it**

- The Segment Radial Gauge is developed from Scale Radial Gauge, using Arc to represent the Segment
- Using both minValue and maxValue in getScale for allowing negative minValue

**- How to verify it**

- Please try example

project2021-code committed:
Added Segment Radial Gauge and make Scale Radial Gauge accept negative minValue

project2021-code committed:
Updated Example

project2021-code committed:
Update Needle startAngle and Example

project2021-code committed:
edit minor difference for PR

project2021-code committed:
edit getScale for negative minValue usage

project2021-code committed:
Test Merge
Merge branch 'trunk' of github.com:project2021-code/at_gauges-segment into trunk

CurtlyCritchlow commented:
@project2021-code, Code looks good, thanks for the PR, We're in the process of merging two of our visualisation packages into one. When that task is completed I'll share the new repo so you can add the changes there.

project2021-code committed:
minor adjustment for PR

project2021-code committed:
allow user to define the major tick decimal place

project2021-code committed:
use SizedBox to define the size, horizontal/vertical will have different
size
adjust the Animation effect
adjust the location of icon
allow user to define the major tick decimal place

CurtlyCritchlow commented:
@project2021-code, Our code have been migrated to https://github.com/atsign-foundation/at_viz/tree/trunk. Please make your PR there, I'm very grateful for your contribution.
